<a href="https://colab.research.google.com/github/anas-git1/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing the CTR/Engagement lane because manually checking every page's click performance doesn't scale with thousands of pages, no one has time to compare each one by hand. If a well-ranked page stays unfixed, that's real traffic and potential sales left on the table, even though the page already earned good visibility. Comparing pages within the same position tier is a fair comparison, because pages at the same position get roughly the same amount of exposure, so a gap in CTR between them isn't about ranking it's about something on the page itself failing to convert that exposure into clicks.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The decision: Which well-ranked pages are under-capturing clicks relative to similar-position pages, and should be reviewed first for a title, meta description, or snippet fix.
Who acts on it: A content editor or SEO specialist they take the top of the ranked list and rewrite the title/meta for those specific pages.
Cost of a wrong call: If the list flags a page that's actually fine (false positive), the editor wastes time on a fix that wasn't needed time that could've gone to a page that genuinely needed help. If the list misses a page that's truly underperforming (false negative), that page keeps silently losing clicks and potential revenue every month, with nobody aware there's a problem.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
import os, subprocess
subprocess.run(["git", "clone", "--depth", "1",
                 "https://github.com/flyrank-bih/flyrank-ml-internship-starter",
                 "flyrank-ml-internship-starter"], check=True)
os.chdir("flyrank-ml-internship-starter")
print("Loaded")

Loaded


In [12]:
import pandas as pd
df = pd.read_csv('data/raw/content_refresh_anonymized.csv')
df.groupby('position_tier')['ctr'].agg(['mean', 'std','count'])

,mean,std,count
position_tier,,,
deep,0.150212,1.612777,1319
page_1,0.652467,3.088921,11814
page_3_5,0.222484,2.147958,7242
striking,0.323239,1.387226,7304
top_3,1.483611,8.187178,2321


In the top_3 position tier, average CTR is 1.48 (in the dataset's scaled units), but the standard deviation is 8.19 — over 5x the mean — showing wide variation in click performance even among similarly-ranked pages. This suggests many top-ranked pages are under-capturing clicks relative to peers at the same position, which is exactly what this lane aims to catch.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This analysis can say: which pages show CTR meaningfully below their position tier's average (an observed pattern), and can rank them to help an editor prioritize review. It cannot say: that fixing a flagged page will definitely increase its clicks, that any CTR change afterward was caused by the fix rather than other factors (seasonality, algorithm updates, competitor changes), or anything about how Google's ranking algorithm actually works. This is decision-support — it points a human to the most promising places to look, not a guarantee of the outcome.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.